src/main/java/com/powsybl/openloadflow/dc/DcLoadFlowEngine.java

```diff
@@ -37,24 +40,22 @@
 
     private final List<LfNetwork> networks;
 
-    private final MatrixFactory matrixFactory;
-
-    private final boolean updateFlows;
-
-    private final boolean useTransformerRatio;
+    private final DcLoadFlowParameters parameters;
 
     public DcLoadFlowEngine(LfNetwork network, MatrixFactory matrixFactory) {
         this.networks = Collections.singletonList(network);
-        this.matrixFactory = Objects.requireNonNull(matrixFactory);
-        this.updateFlows = false;
-        this.useTransformerRatio = true;
+        parameters = new DcLoadFlowParameters(new FirstSlackBusSelector(), matrixFactory, false, true, false, LoadFlowParameters.BalanceType.PROPORTIONAL_TO_GENERATION_P_MAX);
     }
 
     public DcLoadFlowEngine(Object network, DcLoadFlowParameters parameters) {
-        this.networks = LfNetwork.load(network, new LfNetworkParameters(parameters.getSlackBusSelector(), false, false, parameters.isTwtSplitShuntAdmittance(), false));
-        matrixFactory = parameters.getMatrixFactory();
-        updateFlows = parameters.isUpdateFlows();
-        useTransformerRatio = parameters.isUseTransformerRatio();
+        this.networks = LfNetwork.load(network, new LfNetworkParameters(parameters.getSlackBusSelector(), false, false, false, false));
+        this.parameters = Objects.requireNonNull(parameters);
+    }
+
+    private void distributeSlack(LfNetwork network) {
+        double mismatch = network.getActivePowerMismatch();
+        ActivePowerDistribution activePowerDistribution = ActivePowerDistribution.create(parameters.getBalanceType(), false);
+        activePowerDistribution.run(network, -mismatch);
```

Comment:
Maybe we shoud invert the sign of Network.getActivePowerMismatch to avoid this minus anf be consistent everywhere?

Reply:
I have changed the sign. Do you think that I need to perform a functional validation ?

Reply:
I should be ok